### ISSUE TITLE
9GED18000-009 (Weiser Smartcode 10 lock) joins then immediately leaves network

### ISSUE THREAD

**Issue by JasonSanDiego** [Koenkk/zigbee2mqtt#18179]
### What happened?

When pairing this device, the interview and join process goes smoothly and very quickly, but then the device immediately leaves the network and vanishes from the front-end.

### What did you expect to happen?

The device should remain visible and usable on the network.

### How to reproduce it (minimal and precise)

Press pairing button 4 times on device to start pairing. Pairing completes then device leaves.

### Zigbee2MQTT version

1.32.0

### Adapter firmware version

20230507

### Adapter

Sonoff Zigbee 3.0 Plus (ZBDongle-P)

### Debug log

[z2m.log](https://github.com/Koenkk/zigbee2mqtt/files/11928741/z2m.log)

**Comment by JasonSanDiego** [comment]
note - I was running 1.28.0 and original adapter firmware... 202107XX if I remember correctly. I had this problem. I spent the last few hours upgrading z2m and the adapter firmware to latest hoping it would solve it, but no luck. I have seen others comment they had the Smartcode 10 working, but also someone with [Kwikset 910 who had the same issue as me](https://github.com/Koenkk/zigbee2mqtt/discussions/12396).

Happy to do whatever necessary to debug. Feels like it's very close to working!

**Comment by JasonSanDiego** [comment]
I found a workaround for this issue based on reading #14568. Immediately, after seeing interview and configuration completed, I pulled the batteries from the lock. This stopped whatever was causing the device to leave the network and when I put the batteries back in, the lock was still there. All good!

It seems likely that the lock is requesting to join the network twice quickly in succession, and the second time, the coordinator is kicking it off the network. It seems like there could be an opportunity to solve this common problem via the coordinator firmware.

My next issue is that I can't set a pin over z2m, but that's another issue for another day. I posted about it here: https://community.home-assistant.io/t/how-do-you-add-user-codes-to-kwikset-914-via-zigbee2mqtt/453020/14
